aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -69,35 +76,64 @@ public String getPluginKey() {
     public void configure(
             @NonNull JSONObject pluginConfiguration,
             @NonNull Context context
-    ) throws AmplifyException {
+    ) throws AuthException {
         final CountDownLatch latch = new CountDownLatch(1);
+        final AsyncResult<Exception> asyncException = new AsyncResult<>();
 
         AWSMobileClient.getInstance().initialize(
```

Comment:
We had agreed to put this in `configure(...)` since we read the code and it looked like it was only doing configuration.

This current PR is adding initialization logic into the callback, though. We should split that out into the `initialize()` hook.

Reply:
I'll catch up with you on this on a brief call